packages/backend-deployer/src/cdk_error_mapper.ts

```diff
@@ -486,6 +486,16 @@ export class CdkErrorMapper {
       errorName: 'CloudFormationDeploymentError',
       classification: 'ERROR',
     },
+    {
+      errorRegex:
+        /No valid tables found\. Make sure at least one table has a primary key\./,
+      humanReadableErrorMessage:
+        'No valid tables found. Make sure at least one table has a primary key.',
+      resolutionMessage:
+        "If you are using the 'generate schema-from-database' command, make sure the database has at least one table with a primary key.",
+      errorName: 'GenerateSchemaFromDatabaseError',
+      classification: 'ERROR',
+    },
```

Comment:
is `backend-deployer` involved in `generate schema-from-database` command ?

Reply:
https://github.com/aws-amplify/amplify-backend/blob/560878f3af595d4a661f396545fe970fac39504d/packages/schema-generator/src/generate_schema.ts#L45

Reply:
But we should ask data team to throw typed error for this instead of attempting to match strings.

Reply:
In cases like this I think we would want to catch and throw user error in `SchemaGenerator` or wherever makes sense. Unless we want to introduce error mapping for errors that can happen for CLI commands.

Reply:
> In cases like this I think we would want to catch and throw user error in SchemaGenerator

Correct. But it would be best if data team gave us better error to map from.